tests/unit_test/private/fed/app/simulator/simulator_runner_test.py

```diff
@@ -86,3 +86,31 @@ def test_split_names(self, client_names, gpus, expected_split_names):
         runner = SimulatorRunner(job_folder="", workspace="")
         split_names = runner.split_clients(client_names, gpus)
         assert sorted(split_names) == sorted(expected_split_names)
+
+    @pytest.mark.parametrize(
+        "gpus, expected_gpus",
+        [
+            ("[0,1],[1, 2]", ["0,1", "1,2"]),
+            ("[0,1],[3]", ["0,1", "3"]),
+            ("[0,1],[ 3 ]", ["0,1", "3"]),
+        ],
+    )
+    def test_split_gpus_success(self, gpus, expected_gpus):
+        runner = SimulatorRunner(job_folder="", workspace="")
+        split_gpus, _ = runner.split_gpus(gpus)
+        assert split_gpus == expected_gpus
+
+    @pytest.mark.parametrize(
+        "gpus",
+        [
+            "[0,1],3",
+            "0,1,2",
+        ],
+    )
```

Comment:
suggest add test cases of :

`"[]"` => is this valid?
`"[a]"` => is this valid?
`"[01,3]"` => should be invalid
`"[0,1,2]extra"` => should be invalid

Reply:
split_gpus() is just a function to split the gpu list. It does not check / dictate the gpu name format. The gpu name could be potentially anything, depending on each different system. The first 3 cases are valid. The last one is invalid. I will add these into the unit test cases.